src/structures/Widget.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const Base = require('./Base');
+const WidgetUser = require('./WidgetUser');
+const WidgetChannelManager = require('../managers/WidgetChannelManager');
+
+/**
+ * Represents the data about the guild any bot can preview, connected to the specified guild.
+ * @extends {Base}
+ */
+class Widget extends Base {
+  constructor(client, data, id) {
+    super(client);
+
+    if (!data) return;
+    this.id = id;
+    this._patch(data);
+
+    this.members = [];
+    this.channels = new WidgetChannelManager(client);
+  }
+
+  /**
+   * Builds the widget with the provided data.
+   * @param {*} data The raw data of the widget
+   * @private
+   */
+  _patch(data) {
+    if (data?.code !== '50004') {
+      this.disabled = false;
+      this.id = data.id;
+      this.name = data.name;
+      this.instant_invite = data?.instant_invite;
+      if (data.channels) {
+        this.channels.cache.clear();
+        for (const channel of data.channels) this.channels.add(channel);
+      }
+      if (data.members) {
+        this.members = [];
+        for (const user of data.members) this.members.push(new WidgetUser(this.client, user));
+      }
+    } else {
+      delete this.disabled;
+      delete this.name;
+      delete this.instant_invite;
+      delete this.channels;
```

Comment:
I don't think any class in our lib uses `delete` like this, you probably want to set em to null